### ISSUE TITLE
[NEXT-701] [appDir] Vercel providing `params` in `searchParams`

### ISSUE THREAD

**Issue by oBusk** [vercel/next.js#43139]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Wed Mar 2 00:30:59 UTC 2022
Binaries:
  Node: 16.18.1
  npm: 8.19.3
Relevant packages:
  next: 13.0.5-canary.2
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
```

### What browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

When hosting appDir pages on vercel using dynamic segments, the segment `params` are also provided in the `searchParams` prop. This does not happen in dev server `next dev` or when running locally `next build && next start`

| Local Server (`next start`) | Vercel |
|:-:|:-:|
| ![image](https://user-images.githubusercontent.com/13413409/202874076-6841bd87-ee9a-4754-af10-f0703f5f34c4.png) <br/>Segments are in `params` and query is in `searchParams` | ![image](https://user-images.githubusercontent.com/13413409/202874083-4cd57176-40f0-4e06-b5c9-4a2c7896dd0c.png)<br/>Segments are in both `params` and `searchParams` |

### Expected Behavior

Either that local server has same behaviour as Vercel, or Vercel same as local server.

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

https://github.com/oBusk/params-as-searchparams

### To Reproduce

- Create a Next application using `appDir`.
- Host it on Vercel
- SearchParams include more values than expected

<sub>[NEXT-701](https://linear.app/vercel/issue/NEXT-701/[appdir]-vercel-providing-params-in-searchparams)</sub>

**Comment by micmc422** [comment]
Same issue on 13.1.7-canary.8

on dynamic page.tsx (app/[...uri]/page.tsx)
Console.log({params,searchParams})

{
 params: { uri: 'the-rollings-stones-by-mark-seliger-original-silver-print' },
 searchParams: { uri: 'the-rollings-stones-by-mark-seliger-original-silver-print' }
 }
only on vercel, local build and dev works as expected.

**Comment by aralroca** [comment]
Similar issue with `13.1.7-canary.11`, a little bit different but probably related:

## Local `next dev`:

- `/?lang=en`-> `{ params: {}, searchParams: { lang: 'en' } }` 
- `/?lang=es`-> `{ params: {}, searchParams: { lang: 'es' } }` 
- `/?lang=ca`-> `{ params: {}, searchParams: { lang: 'ca' } }` 

## Local `next start`


- `/?lang=en`-> `{ params: {}, searchParams: { lang: 'en' } }` 
- `/?lang=es`-> `{ params: {}, searchParams: { lang: 'en' } }` 
- `/?lang=ca`-> `{ params: {}, searchParams: { lang: 'en' } }` 

Looks the `lang` param is not updated correctly 😅.

In Vercel I didn't try it yet because first I want to fix it locally...

**Comment by aralroca** [comment]
I solved what I commented before adding this export into the page:

```js
export const dynamic = 'force-dynamic';
```

**Comment by ijjk** [comment]
Hi, this has been updated in `v13.2.5-canary.31` of Next.js, please update and give it a try!

**Comment by oBusk** [comment]
| Local Server (`next start`) | Vercel |
|:-:|:-:|
| ![image](https://user-images.githubusercontent.com/13413409/230514939-eaaf2b4d-f5b4-4672-b510-b887ff5cd6eb.png) | ![image](https://user-images.githubusercontent.com/13413409/230514965-cc8cf2c0-f224-4c00-a2f0-37e9a9cd440f.png) |

**Comment by oBusk** [comment]
I can verify that this is fixed with `v13.2.5-canary.31`!

| Local Server (`next start`) | Vercel |
|:-:|:-:|
| ![image](https://user-images.githubusercontent.com/13413409/230514939-eaaf2b4d-f5b4-4672-b510-b887ff5cd6eb.png) | ![image](https://user-images.githubusercontent.com/13413409/230514965-cc8cf2c0-f224-4c00-a2f0-37e9a9cd440f.png) |

https://params-as-searchparams.vercel.app/segment1/segment2?query1=1&query2=2
